common/djangoapps/student/views.py

```diff
@@ -1582,6 +1582,10 @@ def _do_create_account(form, custom_form=None):
                 custom_model = custom_form.save(commit=False)
                 custom_model.user = user
                 custom_model.save()
+
+            if site:
+                # Set UserAttribute indicating the site the user account was created on.
+                UserAttribute.set_user_attribute(user, 'created_on_site', site.domain)
```

Comment:
Initially we will use this data to identify users related to a particular WL partner so that we can sync the user profile data with Salesforce. In general, the site ID is not as meaningful as the site domain in this context. Even if we used the site ID here, there would be no foreign key relationship.